File: scripts/get_traits_from_bety.py
import requests

bety_api = "https://terraref.ncsa.illinois.edu/bety/api/v1/search"
bety_key = "SECRET"


sitename = "Season 6" # Can also specify particular plots here
traits = ["canopy_cover", "canopy_height"]

for t in traits:
    print("Requesting %s" % t)
    full_url = "%s?trait=%s&sitename=~%s&limit=10000&key=%s" % (bety_api, t, sitename, bety_key)
    r = requests.get(full_url, timeout=None)
    if r.status_code == 200:
        print("Writing results to CSV")
        data = r.json()["data"]
        with open("%s %s.csv" % (sitename, t), 'w') as out:
            out.write("date,sitename,trait,description,value\n")
            for entry in data:
                vals = entry["traits_and_yields_view"]
                out.write("%s,%s,%s,%s,%s\n" % (vals["date"],
                                             vals["sitename"],
                                             vals["trait"],
                                             vals["trait_description"],
                                             vals["mean"]))
        print("%s done." % t)

    else:
        print("%s request failed (%s)" % (t, r.status_code))

print("Done.")

File: scripts/get_traits_from_bety.r
# PARTIAL TRAITS LIST
# -------------------
# canopy_height
# canopy_cover
# leaf_length 
# leaf_width
# stalk_diameter_fixed_height
# surface_temperature
# leaf_angle_alpha_src, leaf_angle_beta_src, leaf_angle_alpha_fit, leaf_angle_beta_fit, leaf_chi_src, leaf_chi_fit
# -------------------
trait <- "canopy_cover"

# install.packages('traits')
library(traits)
# get your api key at https://terraref.org/bety/users or ask David
options(betydb_key = readLines('~/.betykey', warn = FALSE),
        betydb_url = "https://terraref.ncsa.illinois.edu/bety/",
        betydb_api_version = 'beta')
trait_data <- betydb_search(trait     = trait,
                            sitename  = "~Season 6",
                            limit     = "none")
readr::write_csv(x = trait_data, path = paste(c('~/season6_', trait, '.csv'), sep=""))
